---
title: "GWAS Problem Set"
output: html_document
---
###R Markdown Tips

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r}
summary(cars)
```

You can also embed plots, for example:

```{r, fig.width=4, fig.height=4, echo=FALSE}
plot(cars)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.

###Problem Set

All required data files are located in the GWAS_workshop.zip folder available at <https://github.com/hwheeler01/GWAS_workshop/blob/master/GWAS_workshop.zip>

CB1908 is a new chemotherapeutic agent developed by CubbyBear PharmaCorp.  Perform a GWAS on the IC<sub>50</sub> of CB1908-treated lymphoblastoid cell lines from the same Yoruba population used in the GWAS workshop (`genotype.*` files).  The phenotypes are located in `CB1908_IC50.txt` Answer the following questions about your analysis and embed relevant code and plots using R Markdown:


1.  Is the CB1908 IC<sub>50</sub> phenotype normally distributed?  Did you adjust the phenotype prior to running your analysis?

```{r}
data <- read.table("~/Desktop/GWAS_workshop/CB1908_IC50.txt",header = T)
data[1:3,]
hist(data$CB1908_IC50,cex.lab=0.7,cex.main=0.7,cex.axis=0.7)

log2_CB1908_IC50 <- log2(data$CB1908_IC50)
hist(log2_CB1908_IC50,cex.lab=0.7,cex.main=0.7,cex.axis=0.7)

shapiro.test(data$CB1908_IC50)

data <- cbind(data,log2_CB1908_IC50)
data[1:3,]
write.table(data,file = "~/Desktop/GWAS_workshop/CB1908_IC50_log2.txt",quote = F,row.names = F)
```
### I did not adjust the phenotype before running the analysis so that I could view the original distribution, which is skewed to the right. After performing the log2 transformation, it became more normally distributed. After performing the Shapiro-Wilk test, the p-value of 0.0005145 suggests that this phenotype is likely not normally distributed as it is less than 0.05. I then added the log2(IC50) phenotype into the data file.

2.	Show the plink command you used to run your GWAS below and explain what each option within the command does.
./plink --bfile genotypes --maf 0.05 --linear --adjust --pheno CB1908_IC50_log2.txt --pheno-name log2_CB1908_IC50 --out CB1908_L2_IC50

### --bfile: tells plink that the datafiles that begin with genotypee are in binary format
    --maf 0.05: filters out any SNPs with a minor allele frequency < 0.05
    --linear: runs a linear additive association test for each SNPs
    --adjust: uses multiple-testing correction methods to adjust p-values for each SNPs
    --pheno: tells plink the phenotype is located in a separate phenotype file
    --pheno-name: tells plink that log2_CB1908_IC50 is the column heading of the phenotype to use in the phenotype file
    --out: tells plink CB1908_L2_IC50 is the text that begins each output file


3.	Make a Q-Q plot and a Manhattan plot of your results.  Embed the plots into this document and write a figure legend that explains each plot.

4.	What does the Q-Q plot tell you about your data?  What potential confounders in your data should you analyze further? (Note: It is not necessary to actually perform the confounder analyses, just explain what you should further investigate.)

5.	Do you have any genome-wide significant hits?  How did you correct for multiple hypothesis testing?  

6.	Is there any functional information known about your top SNPs?  Are they located in or near a gene?  How might these loci be functioning in response to CB1908?  Please use screen captures of appropriate databases or LocusZoom to highlight interesting loci.

7.	Make a plot and figure legend of your top hitâ€™s genotype vs. phenotype.  Is the minor allele associated with resistance or sensitivity to Drug X?

8.	What analyses and experiments would you do next to follow-up on your findings?


